discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -15,11 +16,38 @@
 class IPFSClient:
     """Helper class for Audius Discovery Provider + IPFS interaction"""
 
-    def __init__(self, ipfs_peer_host, ipfs_peer_port):
+    def __init__(
+        self,
+        ipfs_peer_host,
+        ipfs_peer_port,
+        eth_web3=None,
+        shared_config=None,
+        redis=None,
+        eth_abi_values=None,
+    ):
         self._api = ipfshttpclient.connect(
             f"/dns/{ipfs_peer_host}/tcp/{ipfs_peer_port}/http"
         )
-        self._cnode_endpoints = []
+        logger.warning("IPFSCLIENT | initializing")
+
+        # Fetch list of registered content nodes to use during init.
+        # During indexing, if ipfs fetch fails, _cnode_endpoints and user_replica_set are empty
+        # it might fail to find content and throw an error. To prevent race conditions between
+        # indexing starting and this getting populated, run this on init in the instance
+        # in the celery worker
+        if eth_web3 and shared_config and redis and eth_abi_values:
+            self._cnode_endpoints = list(
+                fetch_all_registered_content_nodes(
```

Comment:
since this is always used as a list, perhaps cast it to list in the helper fn itself before returning value?

Reply:
this is actually used as a set to perform a few operations before being cast as a list index_network_peers.py performs some set operations after fetching the data. didn't want to change too much downstream since this was meant to be a copy/paste mostly